lib/shared/store.js

```diff
@@ -151,19 +153,27 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
             data: drive.device
           })
         }
-      }
-
-      const selectedDevice = newState.getIn([ 'selection', 'drive' ])
 
-      if (selectedDevice && !_.find(action.data, {
-        device: selectedDevice
-      })) {
         return storeReducer(newState, {
-          type: ACTIONS.REMOVE_DRIVE
+          type: ACTIONS.REMOVE_DRIVE,
```

Comment:
I still can't figure out why you've got REMOVE_DRIVE both here and on line 170 :-/

Reply:
The first is ensuring the drive should be auto-selected, the other is making sure the selected drives exist since this is an update to the available drives